### PR TITLE
[Tidy] Fix all C99 warnings

### PR DESCRIPTION
**Pull request by jedi22**
In an effort to cleanup the osquery code base, warnings are being addressed. This diff fixes all C99 warnings about allocating dynamic arrays on the stack.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 6abab053315e5a22d2339768f1e1445d0760e9bd (Job results: [1712](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1712/)) failed the code audit or documentation test.

**Comment by theopolis** [comment]
Instead of `malloc`ing these and needing to `free` them later on, could you do either:
# Use `std::vector`'s: http://en.cppreference.com/w/cpp/container/vector/reserve then pass http://en.cppreference.com/w/cpp/container/vector/data to the C-libs?
# Create a `std::shared_ptr` with a dtor that calls `free`?

**Comment by osqueryer** [comment]
:no_good: The commit 546289f06d429162b0e7960c9e0d09cd9b66e689 (Job results: [1732](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1732/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 6afb54d1c01649fe68799dab5d5e15c6d6689750 (Job results: [1733](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1733/)) failed the code audit or documentation test.

File: osquery/tables/networking/darwin/wifi.mm [hunk 1]
```diff
@@ -140,17 +140,15 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
...
     auto count = CFDictionaryGetCount((CFDictionaryRef)networks);
-    auto keys = static_cast<const void **>(malloc(sizeof(void *) * count));
-    auto values = static_cast<const void **>(malloc(sizeof(void *) * count));
-
-    if (keys != nullptr && values != nullptr) {
-      CFDictionaryGetKeysAndValues((CFDictionaryRef)networks, keys, values);
-      for (CFIndex i = 0; i < count; i++) {
-        parseNetworks((CFDictionaryRef)values[i], results);
-      }
+    std::vector<const void *> keys;
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi.mm]: Check out the `3`rd overloaded constructor: http://en.cppreference.com/w/cpp/container/vector/vector

File: osquery/tables/system/darwin/authorizations.mm [hunk 2]
```diff
@@ -131,10 +131,13 @@ QueryData genAuthorizations(QueryContext &context) {
...
-      const void *keys[count];
-      const void *values[count];
-      CFDictionaryGetKeysAndValues(rightSet, keys, values);
-      if ((const void *)keys == nullptr || (const void *)values == nullptr) {
+      std::vector<const void *> keys;
+      keys.resize(count);
+      std::vector<const void *> values;
+      values.resize(count);
+      CFDictionaryGetKeysAndValues(rightSet, keys.data(), values.data());
+      if ((const void *)keys.data() == nullptr ||
```
Comment (theopolis) [inline osquery/tables/system/darwin/authorizations.mm]: Why do you have to explicit cast these? Shouldn't `.data()` be a `const void*` type? http://en.cppreference.com/w/cpp/container/vector/data

**Comment by osqueryer** [comment]
:no_good: The commit 1d9d03448f35ee30f058ec0e21e447d65efd892d (Job results: [1738](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1738/)) failed the code audit or documentation test.

File: osquery/tables/networking/darwin/tests/wifi_tests.mm [hunk 3]
```diff
@@ -37,8 +37,10 @@
   QueryData results;
   auto count = CFDictionaryGetCount(networks);
   ASSERT_EQ((long)count, 2);
-  const void* keys[count];
-  const void* values[count];
+  auto keys = static_cast<const void **>(malloc(sizeof(void *) * count));
```
Comment (theopolis) [inline osquery/tables/networking/darwin/tests/wifi_tests.mm]: Does the `std::vector` method not work here?

**Comment by osqueryer** [comment]
:-1: The commit d620a54809e295389eb46b826a53c5763a201d5c (Job results: [1812](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1812/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit d620a54809e295389eb46b826a53c5763a201d5c (Job results: [1813](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1813/)) failed one or more tests (**Windows**).

**Pull request merged**